src/indexer.cc

```diff
@@ -1255,17 +1255,22 @@ void OnIndexDeclaration(CXClientData client_data, const CXIdxDeclInfo* decl) {
 
       type->def.hover = type->def.detailed_name;
 
-      // For single line Typedef/CXXTypeAlias, display the declaration line,
+      // For Typedef/CXXTypeAlias spanning a few lines, display the declaration line,
       // with spelling name replaced with qualified name.
       // TODO Think how to display multi-line declaration like `typedef struct { ... } foo;`
-      if (extent.start.line == extent.end.line) {
-        std::string decl_text = GetDocumentContentInRange(
-            param->tu->cx_tu, clang_getCursorExtent(decl->cursor));
-        if (decl_text.size() == extent.end.column - extent.start.column) {
-          type->def.hover  =
-              decl_text.substr(0, spell.start.column - extent.start.column) +
+      if (extent.end.line - extent.start.line < 3) {
```

Comment:
pull 3 into a constant.

Reply:
Off-topic: I have seen the exactly same magic number `3` right in the source code of nginx. Just a joke.

Reply:
Done